pkgs/development/tools/build-managers/bazel/bazel-remote/default.nix

```diff
@@ -64,11 +67,19 @@ buildBazelPackage rec {
       sed -e '/^FILE:@bazel_gazelle_go_repository_tools.*/d' -i $bazelOut/external/\@*.marker
     '';
 
-    sha256 = "1m7fmb03lirffxx04ck73bn5zwaji7zdwhlqq8s1c6pgp755d3vi";
+    sha256 = if stdenv.isDarwin then "0h6hmm8kkkwk0q8idlbfr50iplmrm2g7dsy6f5k2hilp8xmfssmw" else "1b6252cz26qwc5nvd34ld3fr908bz55w136x9y27yx39z5ynnbb8";
   };
 
   buildAttrs = {
-    preBuild = ''
+    preBuild = stdenv.lib.optionalString stdenv.isDarwin ''
+        # Framework search paths aren't added by bintools hook
+      # https://github.com/NixOS/nixpkgs/pull/41914
+      export NIX_LDFLAGS+=" -F${CoreFoundation}/Library/Frameworks -F${CoreServices}/Library/Frameworks -F${Foundation}/Library/Frameworks"
+
+      # libcxx includes aren't added by libcxx hook
+      # https://github.com/NixOS/nixpkgs/pull/41589
+      export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -isystem ${libcxx}/include/c++/v1"
```

Comment:
Copied these from the bazel derivation, would be good to move it to `buildBazelPackage`, but changing that requires updating all the shas of the derivations that  use it.

Reply:
there doesn't seem to be that many (a dozen ?) so if you are sure of the fix, it is doable (if you want to :) ).